pinot-controller/src/main/java/org/apache/pinot/controller/ControllerUserDefinedMessageHandlerFactory.java

```diff
@@ -109,6 +113,10 @@ private static Properties createTaskProperties(String periodicTaskRequestId, Str
         periodicTaskParameters.setProperty(PeriodicTask.PROPERTY_KEY_TABLE_NAME, tableNameWithType);
       }
 
+      if (taskParamsJson != null) {
```

Comment:
Here we can write the key-value pairs of the map into the `Properties`

Reply:
Ack